README.md

```diff
@@ -1,2 +1,73 @@
-# Hazel
-Hazel Engine
+# Hazel [![License](https://img.shields.io/github/license/TheCherno/Hazel.svg)](https://github.com/TheCherno/Hazel/blob/master/LICENSE)
+
+## About
+The Hazel Engine is [a 3D graphics game engine Youtube series teaching project](https://thecherno.com/engine/) by Yan "TheCherno" Chernikov. Hazel's goal is to be a library for developers that structures and manages the inner workings of a game, with abstractions to different APIs and platforms.
+
+"TheCherno" is an Australian YouTuber who creates programming tutorials around different subjects from game development to general programming language tutorials at [TheChernoProject](https://www.youtube.com/user/TheChernoProject) on Youtube.
+
+[![Twitter](https://img.shields.io/badge/%40thecherno--blue.svg?style=social&logo=Twitter)](https://twitter.com/thecherno)
+[![Instagram](https://img.shields.io/badge/thecherno--red.svg?style=social&logo=Instagram)](https://www.instagram.com/thecherno)
+[![Youtube](https://img.shields.io/badge/TheChernoProject--red.svg?style=social&logo=youtube)](https://www.youtube.com/user/TheChernoProject)
+[![Discord](https://img.shields.io/badge/TheCherno%20Server--blue.svg?style=social&logo=Discord)](https://discord.gg/K2eSyQA)
+
+## 1. Supported platforms
+Currently Hazel supports:
+
+- Computer OS:
+  - ![Windows supported](https://img.shields.io/badge/Windows-win--64-green.svg)
+  - ![Linux supported](https://img.shields.io/badge/Linux-Debian-green.svg)
+  - ![MacOS not supported](https://img.shields.io/badge/MacOS-Not%20Supported-red.svg)
+- Mobile OS:
+  - ![Android not supported](https://img.shields.io/badge/Android-Not%20Supported-red.svg)
+  - ![IOS not supported](https://img.shields.io/badge/IOS-Not%20Supported-red.svg)
+
+Windows and Linux is currently supported with plans for MacOS and Android/IOS support in the future.
+
+## 2 Installing and setup
+
+Start by cloning the repository with `git clone --recursive <URL>`.
+
+If the repository was cloned non-recursively previously, use `git submodule update --init` to clone the necessary submodules.
+Hazel uses _Premake 5_ as a build generation tool. Visit the [Premake website](https://premake.github.io/download.html) to download and install it.
+
+Next: Follow the steps relevant to your operating system.
+
+### 2.1 Windows
+
+Premake 5.0.0-alpha13 is provided as [premake5.exe](https://github.com/TheCherno/Hazel/blob/master/vendor/bin/premake/premake5.exe) in the repository. Execute and follow the install instructions.
+
+Premake generates project files for Visual Studio. To generate the `.sln` and `.vcxproj` files for Visual Studio 2017, run `premake vs2017` at the command line. Or you may run [GenerateProjects.bat](https://github.com/TheCherno/Hazel/blob/master/GenerateProjects.bat) as a convenience batch file for this task.
```

Comment:
The part where it says 'Visual Studio 2017' should be in my opinion be changed with `Visual Studio 2019`

Reply:
The first sentence states it generates the projects for visual studio without any version (so 2017 and 2019). In the next sentence we give the example of vs2017 and link the GenerateProjects.bat file that cherno is using at this point in time to generate projects for vs2017.

In my opinion it wouldn't make sense to change it to vs2019 while we are currently using it to generate vs2017 files, as cherno is using currently in his videos... In case cherno updates to vs2019 I would change this as well.

Reply:
Visual Studio 2019 should be backward compatible. I wouldn't say it's a fault of the project but of Microsoft for doing these proprietary things ever other year.

Reply:
As this is not realy related to the linux support but rather the windows support, it's not part of this PR, it's just an initial write of a readme, mainly to contain the install instructions for linux as those aren't covered in any cherno videos.

Once this PR is merges, we can create a new PR to update the GenerateProjects.bat to vs2019, including updating the readme about it.

Marking this as resolved.